code/src/moisture_sensor.cpp

```diff
@@ -0,0 +1,16 @@
+#include <moisture_sensor.hpp>
+
+namespace r2d2::plant_rescue_system {
+    moisture_sensor_c::moisture_sensor_c(hwlib::adc &sensor_pin)
+        : sensor_pin(sensor_pin) {
+    }
+
+    uint8_t moisture_sensor_c::get_value() {
```

Comment:
maybe build in a check to see if you get a valid value between your specified min and max values?